moto/iot/models.py

```diff
@@ -1338,7 +1338,7 @@ def list_job_executions_for_thing(
         if status is not None:
             job_executions = list(
                 filter(
-                    lambda elem: status in elem["status"] and elem["status"] == status,
+                    lambda elem: "status" in elem["jobExecutionSummary"] and elem["jobExecutionSummary"]["status"] == status,
```

Comment:
```suggestion
                    lambda elem: elem["jobExecutionSummary"].get("status") == status
```

Reply:
This maybe a  more readable alternative - logically it does the same as far as I can see